libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1191,7 +1203,8 @@ void GCS_MAVLINK::update_send()
 
         // check for any messages that the code has explicitly sent
         const int16_t fs = pushed_ap_message_ids.first_set();
-        if (fs != -1) {
+        // the is_high_latency_link check is to prevent MSG_SYS_STATUS and MSG_HEARTBEAT from AP_Vehicle sneaking in
+        if (fs != -1 && !(is_high_latency_link && (fs == MSG_SYS_STATUS || fs == MSG_HEARTBEAT))) {
```

Comment:
Do we explicitly push `SYS_STATUS`?

This blacklisting makes me somewhat uncomfortable - you've just got two magic message types nominated in here...

... also, this needs a separate method... I can't think of a name for the positive, but:
```suggestion
        // the is_high_latency_link check is to prevent MSG_SYS_STATUS and MSG_HEARTBEAT from AP_Vehicle sneaking in
        if (fs != -1 && !should_filter_pushed_message(fs)) {
```

```
bool GCS_MAVLINK::should_filter_pushed_message(ap_message id) const
{
    if (!is_high_latency_link) {
        return false;
    }
    switch (id) {
    case MSG_HEARTBEAT:
    case MSG_SYS_STATUS:
        return true;
    default:
        return false;
    }
}
```

Reply:
> Do we explicitly push SYS_STATUS?

I did have an issue with AP_Vehicle explicity pushing HEARTBEAT and SYS_STATUS (see https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Vehicle/AP_Vehicle.cpp#L334), but that no longer appears to be a problem. So removed.